### DISCUSSION TITLE
Markdown Support in Step Titles for Clearer Reports

### DISCUSSION THREAD

**Discussion by lararojasmr** [allure-framework/allure2#2593] [Ideas & Feedback]
### Describe the Bug

Summary:
Although not a traditional bug, many users have requested the ability to add markdown to step titles for clearer reporting. Concerns about security and formatting have been addressed with a solution that removes unsafe HTML while allowing safe markdown, ensuring that the report remains intact.

Details:

**Problem:** Users want to include markdown in step titles to enhance report clarity.

- **Solution:** Implement a method to sanitize input by removing unsafe HTML but allowing specific markdown. This ensures the report remains secure and well-formatted.
- **Impact:** This feature will enable users to create more detailed and readable reports without compromising security.

**Example Use Case:**
With this solution, users can now format step titles using markdown, improving the overall readability and organization of their reports.

### Steps to Reproduce

**Test Case: Add and Render Step Titles with Markdown**

**Create a Test Case:** Add a step that includes markdown in the title.
**Render the Report:** Generate the report to review the step titles.
_Actual Report:_
Markdown is not supported in step titles.

_Expected Report:_
Step titles are rendered with markdown processed correctly.

### Expected Behaviour

Step titles are rendered with markdown processed correctly.

### Screenshots or Additional Context

<img width="555" alt="image" src="https://github.com/allure-framework/allure2/assets/15206513/807f9cc2-d83a-4df3-b261-30ce2481e805">

### What Language are you using?

JavaScript

### What Framework/Allure Integration you are using?

allure-commandline, allure-generator, allure-bamboo-plugin

### What version of Allure Integration you are using?

lts

### What version of Allure Report you are using?

lts

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by lararojasmr** [comment]
I've added a solution for this. @baev @vbragin, could you check it? I would like to use this in my projects from now on. Thanks!

**Reply by baev** [reply]
The issue with the proposed solution is that currently, Allure supports over 100 different test frameworks, and most of them provide steps in some form. Sometimes, steps may contain special characters. With your solution, we will break all the existing reports for users, forcing them to update all the integrations (and first, we should provide such an update for officially maintained integrations).

**Comment by baev** [comment]
The good news is that our backlog has a *Markup for steps* feature. The way we see it:

1. Allure 3 (now in the active development stage) will internally support markup in steps. We plan to support bold, italic, underline, strikethrough, links, code and color (both text and background).
2. We will publish a new schema for new Allure results that allows for both marked and plain steps.
3. Then, we'll push updates to language SDKs (commons modules) to support a new results schema
4. And after that, each integration will receive the update with such steps markup feature included
